framework/CodeInterfaces/Prescient/PrescientCodeInterface.py

```diff
@@ -0,0 +1,228 @@
+      line = line.strip()
+      if first:
+        first = False
+        if not line.startswith("Date,Hour,Bus,"): # != "Date,Hour,Bus,Shortfall,Overgeneration,LMP,LMP DA":
+          assert False, "Unexpected first line of bus detail:" + line
+          a = 1/0 #because debug might be disabled
+        dataList = [s.replace(" ","_") for s in line.split(",")[3:]]
+        continue
+      splited = line.split(",")
+      date, hour, bus = splited[:3]
+      rest = splited[3:]
+      key = (date,hour)
+      busSet.add(bus)
+      timeDict = retDict.get(key,{})
+      timeDict[bus] = rest
+      retDict[key] = timeDict
+    busList = list(busSet)
+    busList.sort()
+    return retDict, busList, dataList
+
+  def finalizeCodeOutput(self, command, codeLogFile, subDirectory):
+    """
+      Convert csv information to RAVEN's prefered formats
+      Joins together two different csv files and also reorders it a bit.
+      @ In, command, ignored
+      @ In, codeLogFile, ignored
+      @ In, subDirectory, string, the subdirectory where the information is.
+      @ Out, directory, string, the base name of the csv file
+    """
+    #print("finalizeCodeOutput", command, codeLogFile, subDirectory)
+    toRead = "hourly_summary" #"Daily_summary"
+    if self._output_directory is not None:
+      directory = os.path.join(subDirectory, self._output_directory)
+    else:
+      directory = subDirectory
+    readFile = os.path.join(directory, toRead)
+    if toRead.lower().startswith("hourly"):
+      busData, busList, busDataList = self._readBusData(os.path.join(directory, "bus_detail.csv"))
+      #Need to merge the date and hour
+      readFileNew = readFile + "_merged"
+      outFile = open(readFileNew+".csv","w")
+      inFile = open(readFile+".csv","r")
+      first = True
+      hasNetDemand = False
+      for line in inFile.readlines():
+        date, hour, rest = line.split(",", maxsplit=2)
+        outFile.write(date.rstrip()+"_"+hour.lstrip()+","+rest.rstrip())
+        if first:
+          if ",RenewablesUsed," in rest and ",Demand," in rest:
+            hasNetDemand = True
+            restSplit = rest.split(",")
+            renewablesUsedIndex = restSplit.index("RenewablesUsed")
+            demandIndex = restSplit.index("Demand")
+            outFile.write(","+"NetDemand")
+          first = False
+          for bus in busList:
+            for dataName in busDataList:
+              outFile.write(","+bus+"_"+dataName)
+        else:
+          if hasNetDemand:
+            #Calculate the demand - renewables used to get net demand
+            restSplit = rest.split(",")
+            netDemand = float(restSplit[demandIndex]) - float(restSplit[renewablesUsedIndex])
+            outFile.write(","+str(netDemand))
+          for bus in busList:
+            for data in busData[(date,hour)][bus]:
+              outFile.write(","+data)
+        outFile.write("\n")
+      readFile = readFileNew
+    return readFile
```

Comment:
we do not need to create a CSV for having RAVEN to read it. I would just return the dictionary of data, to avoid double IO

Reply:
Fixed.